sphinx/source/Tutorials/Receiving Messages.rst

```diff
@@ -0,0 +1,196 @@
+.. _ReceivingMessages:
+
+Receiving Messages
+===================
+
+.. toctree::
+   :hidden:
+   :glob:
+
+.. contents:: Contents
+   :depth: 2
+   :local:
+
+So far we've covered setting up the CAN stack, sending broadcast messages, and sending destination specific messages.
+
+Now we'll learn how to receive messages.
+
+Receiving Broadcast Messages
+-----------------------------
+
+Receiving any message from the CAN stack will always come in the form of a callback.
+
+Broadcast messages are the easiest to receive. Simply register a callback for the PGN you want to receive.
+
+In this example, we'll define a function that will process all received proprietary A messages sent to the broadcast address.
+
+.. code-block:: c++
+
+   void propa_callback(isobus::CANMessage *CANMessage, void *parent)
+   {
+   	 if (nullptr != CANMessage)
+     {
+      std::cout << CANMessage->get_data_length() << std::endl;
+     }
+   }
+
+This callback isn't particularly useful, but it does illustrate how to use the callback system.
+
+Basically, whenever a PROPA messsage is received that was sent to the broadcast address, it will print out the length of that message to the console.
```

Comment:
```suggestion
Basically, whenever a PROPA message is received that was sent to the broadcast address, it will print out the length of that message to the console.
```